addons/gearinfo/functions/fnc_findConfigByValue.sqf

```diff
@@ -0,0 +1,24 @@
+#include "script_component.hpp"
+
+params ["_classRoot", "_model", "_optionSet"];
+_optionSet params ["_optionIndex", "_valueName"];
+
+private _firstMatch = GVAR(weakMatchesCache) getOrDefault [_optionSet, configNull];
```

Comment:
`_optionSet` will collide between models, the key should include also `_classRoot` and `_model`

Reply:
I just extended it with `_classRoot` and `_model` now, without a deep copy of `_optionSet` for performance sake. Should be fine, shouldn't it?

Reply:
Yes, it's fine. Deep copy is not required as array is not modified after, anyway engine make itself a deep copy of key on insert ("Supported Key Types" https://community.bistudio.com/wiki/HashMap).